mapie/mondrian.py

```diff
@@ -0,0 +1,419 @@
+    ----------
+    unique_groups : NDArray
+        The unique groups of individuals for which the estimator was fitted
+
+    mapie_estimators : Dict
+        A dictionary containing the fitted conformal estimator for each group
+        of individuals
+
+    References
+    ----------
+    Vladimir Vovk, David Lindsay, Ilia Nouretdinov, and Alex Gammerman.
+    Mondrian confidence machine.
+    Technical report, Royal Holloway University of London, 2003
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.linear_model import LogisticRegression
+    >>> from mapie.classification import MapieClassifier
+    >>> X_toy = np.arange(9).reshape(-1, 1)
+    >>> y_toy = np.stack([0, 0, 1, 0, 1, 2, 1, 2, 2])
+    >>> groups_toy = [0, 0, 0, 0, 1, 1, 1, 1, 1]
+    >>> clf = LogisticRegression(random_state=42).fit(X_toy, y_toy)
+    >>> mapie = MondrianCP(MapieClassifier(estimator=clf, cv="prefit")).fit(
+    ...     X_toy, y_toy, groups_toy)
+    >>> _, y_pi_mapie = mapie.predict(X_toy, alpha=0.4, groups=groups_toy)
+    >>> print(y_pi_mapie[:, :, 0].astype(bool))
+    [[ True False False]
+     [ True False False]
+     [ True  True False]
+     [ True  True False]
+     [False  True False]
+     [False  True  True]
+     [False False  True]
+     [False False  True]
+     [False False  True]]
+    """
+
+    not_allowed_estimators = (
+        MapieCalibrator,
+        MapieMultiLabelClassifier,
+        MapieQuantileRegressor,
+        MapieTimeSeriesRegressor
+    )
+    allowed_classification_ncs_str = [
+        "lac", "score", "cumulated_score", "aps", "top_k"
+    ]
+    allowed_classification_ncs_class = (
+        LACConformityScore, NaiveConformityScore, APSConformityScore,
+        TopKConformityScore
+    )
+    allowed_regression_ncs = (
+        AbsoluteConformityScore, GammaConformityScore
+    )
+    fit_attributes = [
+        "unique_groups",
+        "mapie_estimators"
+    ]
+
+    def __init__(
+        self,
+        mapie_estimator: Union[MapieClassifier, MapieRegressor]
+    ):
+        self.mapie_estimator = mapie_estimator
+
+    def fit(
+        self, X: ArrayLike,
+        y: ArrayLike,
+        groups: ArrayLike,
+        **fit_params
+    ) -> MondrianCP:
+        """
+        Fit the Mondrian method
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            The input data
+
+        y : ArrayLike of shape (n_samples,) or (n_samples, n_outputs)
+            The target values
+
+        groups : ArrayLike of shape (n_samples,)
+            The groups of individuals. Must be defined by integers. There must
+            be at least 2 individuals per group.
+
+        **fit_params
+            Additional keyword arguments to pass to the estimator's fit method
+            that may be specific to the Mapie estimator used
+        """
+
+        X, y, groups = self._check_fit_parameters(X, y, groups)
+        self._check_group_length(X, groups)
```

Comment:
Why isn't this done in the `_check_fit_parameters`?

Reply:
done